state.go

```diff
@@ -963,7 +968,7 @@ func (m *Memberlist) aliveNode(a *alive, notify chan struct{}, bootstrap bool) {
 				time.Since(state.StateChange) > m.config.DeadNodeReclaimTime)
 
 			// Allow the address to be updated if a dead node is being replaced.
-			if state.State == stateDead && canReclaim {
+			if state.State == stateLeft || (state.State == stateDead && canReclaim) {
 				m.logger.Printf("[INFO] memberlist: Updating address for failed node %s from %v:%d to %v:%d",
```

Comment:
Should we change the log message in the new case? It's not really right any more as the node didn't "fail". Not a big deal but might be clearer?

Reply:
updated the log to mention that the node might have left as well.